Notify new links

1s
Run # Get the diff for links.csv files
  # Get the diff for links.csv files
  DIFF=$(git diff HEAD~1 HEAD -- 'links/**/links.csv')
  
  if [ -z "$DIFF" ]; then
    echo "No changes to links.csv files"
    exit 0
  fi
  
  # Extract removed lines (old values) - exclude header and diff markers
  REMOVED=$(echo "$DIFF" | grep '^-' | grep -v '^---' | grep -v '^-slug,url' | cut -c2- || true)
  
  # Extract added lines (new values) - exclude header and diff markers
  ADDED=$(echo "$DIFF" | grep '^+' | grep -v '^+++' | grep -v '^+slug,url' | cut -c2- || true)
  
  if [ -z "$ADDED" ]; then
    echo "No new or modified links found"
    exit 0
  fi
  
  # Build arrays of slugs from removed lines for modification detection
  declare -A REMOVED_SLUGS
  while IFS=',' read -r slug url; do
    [ -n "$slug" ] && REMOVED_SLUGS["$slug"]="$url"
  done <<< "$REMOVED"
  
  # Categorize links as NEW or MODIFIED
  NEW_LINKS=""
  MODIFIED_LINKS=""
  
  while IFS=',' read -r slug url; do
    [ -z "$slug" ] && continue
    
    if [ -n "${REMOVED_SLUGS[$slug]:-}" ]; then
      # Slug existed before - this is a modification
      OLD_URL="${REMOVED_SLUGS[$slug]}"
      if [ "$OLD_URL" != "$url" ]; then
        MODIFIED_LINKS+="- [gitly.sh/$slug](https://gitly.sh/$slug) → $url
    "
      fi
    else
      # Slug is new
      NEW_LINKS+="- [gitly.sh/$slug](https://gitly.sh/$slug) → $url
    "
    fi
  done <<< "$ADDED"
  
  # Build comment body
  COMMENT=""
  
  if [ -n "$NEW_LINKS" ]; then
    COMMENT+="## 🎉 New links are live!
  
  $NEW_LINKS
  "
  fi
  
  if [ -n "$MODIFIED_LINKS" ]; then
    COMMENT+="## 🔄 Modified links updated!
  
  $MODIFIED_LINKS
  "
  fi
  
  if [ -z "$COMMENT" ]; then
    echo "No new or modified links to report"
    exit 0
  fi
  
  # Post comment to PR
  gh pr comment $PR_NUMBER --body "$COMMENT"
  shell: /usr/bin/bash -e {0}
  env:
    GH_TOKEN: ***
    PR_NUMBER: 144
GraphQL: Resource not accessible by integration (addComment)
Error: Process completed with exit code 1.

--- a/.github/workflows/post-merge-notify.yml
+++ b/.github/workflows/post-merge-notify.yml
@@ -7,6 +7,9 @@ on:
     paths:
       - "links/**/*.csv"
 
+permissions:
+  pull-requests: write
+
 jobs:
   notify-new-links:
     if: github.event.pull_request.merged == true

